### PR TITLE
refactor: algin plugin guide component

### PR DESCRIPTION
**Pull request by Lanttcat**
## Description

Closes #MF-2215
## Type of change

- [ ] Documentation
- [x] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

## Checklist

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
  - [ ] I have removed all in development `console.log`s
  - [ ] I have removed all commented code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

If this PR depends on external APIs:

- [ ] I have configured those APIs with CORS headers to let extension requests get passed. 
  - chrome extension: `chrome-extension://[id]`
  - firefox extension: `moz-extension://[id]`
- [ ] I have delegated all web requests to the background service via the internal RPC bridge.

### CODE REVIEW

File: packages/shared/src/UI/components/PluginGuide/usePluginGuideRecord.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+    const nextStep = useCallback(() => {
+        if (!storage) return
+
+        storage.userGuide.setValue({ ...record, [key ?? 'default']: currentStep + 1 })
+    }, [storage, record, currentStep])
+
+    useEffect(() => {
+        if (currentStep !== totalStep) return
+        onFinish?.()
```
Comment (guanbinrui) [inline packages/shared/src/UI/components/PluginGuide/usePluginGuideRecord.ts]: To call it in `nextStep`.

File: packages/mask/src/plugins/Tips/SNSAdaptor/components/TipsRealmContent/index.tsx [hunk 2]
```diff
@@ -89,11 +93,24 @@ export const TipsRealmContent: Plugin.InjectUI<Plugin.SNSAdaptor.TipsRealmOption
...
-            <Guide>
-                <Stack display="inline-block" width="38px" height="38px" position="relative" top={2}>
-                    {button}
-                </Stack>
-            </Guide>
+            <PluginGuideProvider
+                value={{
+                    storageKey: EnhanceableSite.Mirror,
+                    totalStep: 1,
+                    storage: getStorage(),
```
Comment (guanbinrui) [inline packages/mask/src/plugins/Tips/SNSAdaptor/components/TipsRealmContent/index.tsx]: Why not create storage in `PluginGuideProvider`?

File: packages/mask/src/plugins/Tips/SNSAdaptor/components/TipsRealmContent/index.tsx [hunk 3]
```diff
@@ -89,11 +93,24 @@ export const TipsRealmContent: Plugin.InjectUI<Plugin.SNSAdaptor.TipsRealmOption
...
         return (
-            <Guide>
-                <Stack display="inline-block" width="38px" height="38px" position="relative" top={2}>
-                    {button}
-                </Stack>
-            </Guide>
+            <PluginGuideProvider
+                value={{
+                    storageKey: EnhanceableSite.Mirror,
+                    totalStep: 1,
```
Comment (guanbinrui) [inline packages/mask/src/plugins/Tips/SNSAdaptor/components/TipsRealmContent/index.tsx]: You can compute it from `guides.length`.

File: packages/mask/src/plugins/Tips/SNSAdaptor/components/TipsRealmContent/index.tsx (hunk 2 above)
Comment (Lanttcat) [reply packages/mask/src/plugins/Tips/SNSAdaptor/components/TipsRealmContent/index.tsx]: I think the plugin can decides the storage provider.  The storage provider can be storage service or global variable. And plugin also should  keep and manage this value.

File: packages/mask/src/plugins/Tips/SNSAdaptor/components/TipsRealmContent/index.tsx (hunk 3 above)
Comment (Lanttcat) [reply packages/mask/src/plugins/Tips/SNSAdaptor/components/TipsRealmContent/index.tsx]: The `totalStep` is clear and understandable for developer.  I think should keep it.

File: packages/mask/src/plugins/Tips/SNSAdaptor/components/TipsRealmContent/index.tsx (hunk 2 above)
Comment (guanbinrui) [reply packages/mask/src/plugins/Tips/SNSAdaptor/components/TipsRealmContent/index.tsx]: The primary goal of a shared component is to reduce repeating working. No option is better than too many options.

File: packages/mask/src/plugins/Tips/SNSAdaptor/components/TipsRealmContent/index.tsx (hunk 3 above)
Comment (guanbinrui) [reply packages/mask/src/plugins/Tips/SNSAdaptor/components/TipsRealmContent/index.tsx]: No need to let the user of `<PluginGuideProvider />` to concern about it.

**Review by guanbinrui — APPROVED** [review]

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**